sktime/transformations/panel/window_summarizer.py

```diff
@@ -0,0 +1,268 @@
+                lambda x: getattr(x.shift(shift).rolling(win), win_summarizer)()
+            )
+    else:
+        feat = Z.shift(shift)
+        if isinstance(Z.index, pd.MultiIndex) and callable(win_summarizer):
+            feat = feat.rolling(win).apply(win_summarizer, raw=True)
+        elif not isinstance(Z.index, pd.MultiIndex) and callable(win_summarizer):
+            feat = feat.apply(lambda x: x.rolling(win).apply(win_summarizer, raw=True))
+
+    if isinstance(feat, pd.Series):
+        feat = pd.DataFrame(feat)
+
+    feat.rename(
+        columns={
+            feat.columns[0]: name + "_" + "_".join([str(item) for item in window])
+        },
+        inplace=True,
+    )
+
+    return feat
+
+
+class _LaggedWindowExtractor(_PanelToTabularTransformer):
+    """Base adapter class for transformations.
+
+    The LaggedWindowSummarizer transforms input series to features
+    based on a provided dictionary of window summarizer, window shifts
+    and window lengths.
+    """
+
+    _tags = {
+        # todo: what is the scitype of X: Series, or Panel
+        # "scitype:transform-input": "Panel",
+        # todo: what scitype is returned: Primitives, Series, Panel
+        "scitype:transform-output": "Panel",
+        # todo: what is the scitype of y: None (not needed), Primitives, Series, Panel
+        "scitype:transform-labels": "None",
+        "scitype:instancewise": True,  # is this an instance-wise transform?
+        # "X_inner_mtype": ["pd.DataFrame"
+        # # which mtypes do _fit/_predict support for X?
+        # # X_inner_mtype can be Panel mtype even if
+        # transform-input is Series, vectorized
+        # "y_inner_mtype": "None",  # which mtypes do _fit/_predict support for y?
+        "capability:inverse_transform": False,  # does transformer
+        # have inverse transform
+        "skip-inverse-transform": False,  # is inverse-transform skipped when called?
+        "univariate-only": False,  # can the transformer handle multivariate X?
+        "handles-missing-data": True,  # can estimator handle missing data?
+        "X-y-must-have-same-index": True,  # can estimator handle different X/y index?
+        "enforce_index_type": None,  # index type that needs to be enforced in X/y
+        "fit-in-transform": False,  # is fit empty and can be skipped? Yes = True
+        "transform-returns-same-time-index": False,
+        # does transform return have the same time index as input X
+    }
+
+    def __init__(
+        self,
+        functions=None,
+        n_jobs=-1,
+        target_cols=None,
+    ):
+
+        self.functions = functions
+        self.n_jobs = n_jobs
+        self.target_cols = target_cols
+
+        super(_LaggedWindowExtractor, self).__init__()
+
+    # Get extraction parameters
+    def _fit(self, X, y=None):
+        """Fit transformer to X and y.
+
+        Private _fit containing the core logic, called from fit
+        Parameters
+        ----------
+        X : Series or Panel
```

Comment:
change this to the type expected internally

Reply:
Not sure, the transform can take Series and Pandas DataFrame as inputs and will return a Pandas Dataframe (typically one Series / column gets transformed to multiple columns). Panel is not supported due to the changes required that I factored out (part of hierarchical PR). So what should I choose here?

Reply:
this is the docstring of the inner function, so it should say
`X : pandas.DataFrame or pandas.Series`
and `pandas.DataFrame` for the return object.